src/net/websocketsIO.cpp

```diff
@@ -134,5 +140,13 @@ bool WebsocketsClient::wsIsConnected()
     }
     
     assert (thread_id == pthread_self());
-    return ctx->wsIsConnected();
+    bool ret = ctx->wsIsConnected();
```

Comment:
These changes (for `libwebsockets`) are focused to avoid a `wsDisconnect()` call when the socket has been closed by server. First, the disconnect attempt will attempt to gracefully disconnect by sending data over the socket, which results on an error message from the library. Second, the next `wsConnect()` call will fail in debug mode because the context (`ctx`) is still valid. So here we discard the context if no socket.

Reply:
With the changes in the callback of `libwebsockets` in this pull request, the first problem should be fixed, because the external disconnection seems to be correctly processed. 

The second problem could be fixed more reliably by deleting `ctx` there (in `wsConnect`). It would be safer because there isn't any guarantee of receiving a call to this function before a `wsConnect`. If the app manages immediate disconnects and disconnect callbacks correctly, it shouldn't be needed at all.